src/api/apiClient.ts

```diff
@@ -11,16 +11,28 @@ const instance = axios.create({
   },
 });
 
+const getAuthToken = async (): Promise<string | undefined> => {
+  // attempt to retrieve the current session and get the auth token
+  // If an exception occurs, the session has expired. Force the user to authenticate
+  try {
+    const token = (await Auth.currentSession()).getIdToken().getJwtToken();
+
+    return token;
+  } catch {
+    await Auth.federatedSignIn({
+      customProvider: "IdP",
+    });
+  }
+};
+
 // Handle adding the authorization header based on the current session
 instance.interceptors.request.use(async (config) => {
-  // currentSession() should refresh the session if required in most cases
-  // if that turns out to not be the case, we'll need to refresh the token
-  // manually.
-  const token = (await Auth.currentSession()).getIdToken().getJwtToken();
+  const token = await getAuthToken();
   config.headers.Authorization = `Bearer ${token}`;
```

Comment:
```suggestion
  if (token) {
    config.headers.Authorization = `Bearer ${token}`;
  }
```

Since `getAuthToken` can _technically_ return `undefined`, can we do this just to avoid setting the header to an invalid value when we've kicked the user over to the login page?